src/Sylius/Behat/Context/Api/Shop/CartContext.php

```diff
@@ -62,6 +62,8 @@ public function iSeeTheSummaryOfMyCart(string $tokenValue): void
     }
 
     /**
+     * @Given /^I added (product "[^"]+") to the (cart)$/
+     * @Given /^I (?:add|added) ("[^"]+" product) to the (cart)$/
```

Comment:
The rule of thumb is to not mix `given` and `when` scenarios. Can you extract past tense definitions to `Setup\CartContext` (with the usage of commands instead of API calls)

Reply:
It is not easy because is many differences with using this steps in @ui and @api tests, First problem is properly token transfer between ui steps, because now in @ui context we open product by slug and add it to cart, next steps are continue with this cart, its difficult operate with token from setup context step